doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,241 @@
+## `GradBoosting`
+
+The `GradBoosting` class implements a gradient boosting model that supports
+numerical and categorical features. The class offers several template 
+parameters and several runtime options that can be used to control the 
+model.
+
+Gradient Boosting is a very powerful ensemble algorithm used for both 
+classification and regression tasks. It utilizes a series of weak learners 
+(specifically [decision tree regressors](decision_tree_regressor.md)) to arrive 
+closer and closer to a targeted label. Each 
+weak learner is trained on the error of the previous learner, thereby 
+reducing the error of the overall model.
+
```

Comment:
Can we add a bit about the fact that gradient boosting is for classification here?  Here is what's used in the AdaBoost documentation (and other classifiers):

```
`AdaBoost` is useful for classifying points with _discrete labels_ (i.e. `0`,
`1`, `2`).
```